File: HW5_b/Part 2/Fruit.cpp
#include "Fruit.h"

Fruit::Fruit(): Food()
{
	name = "";
}

Fruit::Fruit(time_t r, string n): Food(name)
{
	name = n;
	ripe = r;
}

void Fruit::prepare()
{
	cout << "You can eat it on: " << ripe << endl;
}

string Fruit::getName()
{
	return name;
}

time_t Fruit::getDate()
{
	return ripe;
}

File: HW5_b/Part 1/Fruit.java
import java.util.*;

public class Fruit extends Food implements Comparable<Fruit>
{
	public Date ripe;
	
	public Fruit()
	{
		name = "";
	}
	
	public Fruit( Date r, String n)
	{
		name = n;
		ripe = r;
	}
	
	public void ripeTime()
	{
		System.out.println("You can eat it on " + ripe);
	}
	
	public String getName()
	{
		return name;
	}
	
	public Date getDate()
	{
		return ripe;
	}

	@Override
	public int compareTo(Fruit a) 
	{
		return this.getDate().compareTo(a.getDate());
	}
}